fix URL signing

rjbs opened the pull request:
This will make tests (in another branch) pass.

rjbs committed:
URL signing: sign the digest of the URL

...not the URL

rjbs committed:
URL signing: clip signed time to latest hour

... to be nice to caching

mmcclimon reviewed (commented):
I think the result is fine but I would squash commits 1 and 3 (57be42c5 and 787ab0df, at time of writing.

rjbs committed:
URL signing: punt on exactly what to sign

some systems (notably Fastmail and Topicbox) have slightly different
signature targets
